src/PIL/TiffImagePlugin.py

```diff
@@ -1416,7 +1421,16 @@ def _setup(self) -> None:
         if not isinstance(xsize, int) or not isinstance(ysize, int):
             msg = "Invalid dimensions"
             raise ValueError(msg)
-        self._size = xsize, ysize
+        self._will_be_transposed = self.tag_v2.get(ExifTags.Base.Orientation) in (
+            5,
+            6,
+            7,
+            8,
+        )
+        if self._will_be_transposed:
+            self._size = ysize, xsize
```

Comment:
Please be aware of following `self.size` using in the same function, which should be fixed:

https://github.com/python-pillow/Pillow/pull/8387/files#diff-6ad43f85f1a075181d4d8cfcd97ae27bba1eccf5c3db5a3457160f98218eba6eR1569
https://github.com/python-pillow/Pillow/pull/8387/files#diff-6ad43f85f1a075181d4d8cfcd97ae27bba1eccf5c3db5a3457160f98218eba6eR1603-R1605

Reply:
I've pushed a commit for this.